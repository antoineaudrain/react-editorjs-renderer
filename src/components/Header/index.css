.er-header {
    font-family: Lucida Grande,Lucida Sans Unicode,Lucida Sans;
    letter-spacing: -.03em;
    padding: 1em 0;
    margin: 0 0 -0.9em 0;
    line-height: 1.5em;
    outline: none;
    font-size: 15px;
    color: #313649;
}