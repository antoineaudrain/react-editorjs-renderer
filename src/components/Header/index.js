import React from 'react'
import PropTypes from 'prop-types'
import propsConstructor from '../../tools/propsConstructor'
import defaultStyle from './index.style'

Header.propTypes = {
  data: PropTypes.shape({
    text: PropTypes.string.isRequired,
    level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]).isRequired
  }).isRequired,
  style: PropTypes.object
}

function Header({ data, style = {} }) {
  const getProps = propsConstructor(style, defaultStyle)
  const headerStyle = getProps('header')

  const content = { __html: data.text }

  switch (data.level) {
    case 1:
      return <h1 {...headerStyle} dangerouslySetInnerHTML={content} />
    case 2:
      return <h2 {...headerStyle} dangerouslySetInnerHTML={content} />
    case 3:
      return <h3 {...headerStyle} dangerouslySetInnerHTML={content} />
    case 4:
      return <h4 {...headerStyle} dangerouslySetInnerHTML={content} />
    case 5:
      return <h5 {...headerStyle} dangerouslySetInnerHTML={content} />
    default:
      return ''
  }
}

export default Header
