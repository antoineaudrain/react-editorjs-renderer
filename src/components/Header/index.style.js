const style = {
  fontFamily: 'Lucida Grande,Lucida Sans Unicode,Lucida Sans',
  letterSpacing: '-.03em',
  padding: '1em 0',
  margin: '0 0 -0.9em 0',
  lineHeight: '1.5em',
  outline: 'none',
  color: '#313649'
}

export default style