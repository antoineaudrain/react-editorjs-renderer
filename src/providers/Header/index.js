import React from 'react'
import PropTypes from 'prop-types'
import ReactHtmlParser from 'react-html-parser'
import providerPropsConstructor from '../../utils/providerPropsConstructor'
import defaultStyle from './index.style'

const Header = ({ data, style = {} }) => {
  const getProps = providerPropsConstructor(defaultStyle, style)
  const headerStyle = getProps('header')

  const content = ReactHtmlParser(data.text)

  switch (data.level) {
    case 1:
      return <h1 {...headerStyle}>{content}</h1>
    case 2:
      return <h2 {...headerStyle}>{content}</h2>
    case 3:
      return <h3 {...headerStyle}>{content}</h3>
    case 4:
      return <h4 {...headerStyle}>{content}</h4>
    case 5:
      return <h5 {...headerStyle}>{content}</h5>
    default:
      return ''
  }
}

Header.propTypes = {
  data: PropTypes.shape({
    text: PropTypes.string.isRequired,
    level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]).isRequired
  }).isRequired,
  style: PropTypes.object
}

export default Header
