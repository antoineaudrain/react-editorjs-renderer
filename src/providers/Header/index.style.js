const style = {
  header: {
    letterSpacing: '-.03em',
    padding: '1em 0',
    margin: 0,
    marginBottom: '-0.9em',
    lineHeight: '1.5em',
    outline: 'none'
  }
}

export default style
