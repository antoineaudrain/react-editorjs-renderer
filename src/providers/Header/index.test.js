import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Header from './index'

configure({ adapter: new Adapter() })

describe('Header provider:', () => {
  const headerH1 = shallow(<Header data={{ text: 'test', level: 1 }} />)
  const headerH2 = shallow(<Header data={{ text: 'test', level: 2 }} />)
  const headerH3 = shallow(<Header data={{ text: 'test', level: 3 }} />)
  const headerH4 = shallow(<Header data={{ text: 'test', level: 4 }} />)
  const headerH5 = shallow(<Header data={{ text: 'test', level: 5 }} />)
  const headerH6 = shallow(<Header data={{ text: 'test', level: 6 }} />)

  it('should match snapshot', () => expect(headerH1).toMatchSnapshot())
  it('should match snapshot', () => expect(headerH2).toMatchSnapshot())
  it('should match snapshot', () => expect(headerH3).toMatchSnapshot())
  it('should match snapshot', () => expect(headerH4).toMatchSnapshot())
  it('should match snapshot', () => expect(headerH5).toMatchSnapshot())
  it('should match snapshot', () => expect(headerH6).toMatchSnapshot())
})
