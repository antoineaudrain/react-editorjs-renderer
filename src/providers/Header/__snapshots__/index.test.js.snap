// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header provider: should match snapshot 1`] = `ShallowWrapper {}`;

exports[`Header provider: should match snapshot 2`] = `ShallowWrapper {}`;

exports[`Header provider: should match snapshot 3`] = `ShallowWrapper {}`;

exports[`Header provider: should match snapshot 4`] = `ShallowWrapper {}`;

exports[`Header provider: should match snapshot 5`] = `ShallowWrapper {}`;

exports[`Header provider: should match snapshot 6`] = `ShallowWrapper {}`;
